src/newgrf.cpp

```diff
@@ -391,9 +391,8 @@ void CDECL grfmsg(int severity, const char *str, ...)
  */
 static GRFFile *GetFileByGRFID(uint32 grfid)
 {
-	const GRFFile * const *end = _grf_files.End();
-	for (GRFFile * const *file = _grf_files.Begin(); file != end; file++) {
-		if ((*file)->grfid == grfid) return *file;
+	for (GRFFile * const file : _grf_files) {
+		if (file->grfid == grfid) return file;
```

Comment:
std::find_if? I guess it might not be any shorter.

Reply:
I found find_if to be a bit  more cumbersome before C++14 reduced the bolier-plate, perhaps in a future PR?